public/app/features/explore/state/explorePane.ts

```diff
@@ -109,6 +116,7 @@ interface InitializeExplorePayload {
   datasourceInstance?: DataSourceApi;
   compact: boolean;
   eventBridge: EventBusExtended;
+  queryLibraryRef?: string;
```

Comment:
Thanks for removing it from the URL sync  🙌  

It's still being saved into Explore's state though, not part of `DataQuery` ([discussion](https://github.com/grafana/grafana/pull/108326/files#r2236070650)). This requires now importing and calling internal Explore's state functions in Enterprise :/

```
import { updateQueryLibraryRefAction } from 'app/features/explore/state/explorePane';
```

Any reason why adding new prop to `DataQuery` didn't work out?

Reply:
@ifrost, is there any reason we would prefer to add a new property to that interface used everywhere?
Or you mean creating a new interface that extends from `DataQuery`? 
How should the flow be in that case? 
If I'm not wrong, with the enterprise PR, we [create the explore URL](https://github.com/grafana/grafana-enterprise/pull/9028/files#diff-57ef6434e1482d65a9dbf3d5944e458b53a69cc8cf112fb8ac6131552ef060d7R156) with the specific query. should that query also have a new property and then be parsed in the explore scope?
Is it wrong to import an Explore function in Enterprise? Although explore is an OSS feature, there are many imports from OSS to Enterprise code